[ROCm] Enable inductor-periodic testing for MI300

BLOrange-AMD opened the pull request:
cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

BLOrange-AMD committed:
Initial change to update yml file

BLOrange-AMD committed:
Updated with MI300 gpu name

huydhn commented on .github/workflows/inductor-periodic.yml:
```diff
@@ -78,6 +78,43 @@ jobs:
+
+  linux-focal-rocm6_3-py3_10-periodic-dynamo-benchmarks-test:
+    name: rocm6_3-py3_10-periodic-dynamo-benchmarks
+    uses: ./.github/workflows/_linux-test.yml
```
Just curious, this is using `_linux_test` workflow instead of the regular `_rocm_test` workflow.  Is this expected?

BLOrange-AMD replied:
@huydhn Fixed the issue. Thanks for pointing that out.

BLOrange-AMD committed:
Fixed format issues and updated with rocm_test

jeffdaily committed:
update ROCm Dockerfile for inductor benchmarks

huydhn commented on .github/workflows/inductor-periodic.yml:
```diff
@@ -78,6 +78,44 @@ jobs:
+        ]}
+    secrets: inherit
+
+  linux-focal-rocm6_3-py3_10-periodic-dynamo-benchmarks-test:
```
I think you will need the following snippet to get the necessary permission to run this job https://github.com/pytorch/pytorch/blob/main/.github/workflows/trunk.yml#L185-L187

```
permissions:
  id-token: write
  contents: read
```

jeffdaily committed:
don't build native for ucc, use latest tip

BLOrange-AMD committed:
Added permissions to test running

jeffdaily commented:
@pytorchbot rebase

jeffdaily committed:
rocm-specific logic for fbgemm, expected accuracy

- needed gcc 11 for fbgemm build
- custom build command for fbgemm
- copy all exepected accuracy csv files into rocm dir, update paths

jeffdaily committed:
update ci_expected_accuracy

jeffdaily committed:
fbgemm needs patchelf

jeffdaily committed:
fix copy/paste mistake

BLOrange-AMD committed:
Fixed lint issue

BLOrange-AMD committed:
Updated ci_expected_accuracy for ROCm

BLOrange-AMD commented:
@pytorchbot rebase

jeffdaily committed:
Updated ci_expected_accuracy for ROCm

huydhn commented on .github/workflows/inductor-periodic.yml:
```diff
@@ -67,6 +67,47 @@ jobs:
       test-matrix: ${{ needs.linux-focal-cuda12_4-py3_10-gcc9-periodic-dynamo-benchmarks-build.outputs.test-matrix }}
     secrets: inherit
 
+  linux-focal-rocm6_3-py3_10-periodic-dynamo-benchmarks-build:
```
Sorry, I should have take a closer look at the PR sooner, but the new benchmark jobs needs to be added to https://github.com/pytorch/pytorch/blob/main/.github/workflows/inductor-perf-test-nightly.yml instead.  There is nothing wrong with the current batch of jobs in inductor-periodic, but they only run accuracy test for torch.compile vs eager , which are not those powering https://hud.pytorch.org/benchmark/compilers dashboard.

Let me update the PR a bit

huydhn replied:
Here is my test PR https://github.com/pytorch/pytorch/pull/145546 and the test run is at https://github.com/pytorch/pytorch/actions/runs/12939147191

jeffdaily committed:
add flaky rocm models to check_accuracy.py

BLOrange-AMD committed:
Fixed lintrunner issue

BLOrange-AMD committed:
Check rocm in file name path

BLOrange-AMD committed:
Updated graph_breaks targets

BLOrange-AMD committed:
Corrected targets

jeffdaily committed:
missing entries in csv files for check_graph_breaks.py

BLOrange-AMD commented:
@pytorchbot rebase

BLOrange-AMD committed:
Trigger nightly jobs

malfet commented on .ci/docker/build.sh:
```diff
@@ -86,6 +86,10 @@ CMAKE_VERSION=3.18.5
 
 _UCX_COMMIT=7bb2722ff2187a0cad557ae4a6afa090569f83fb
 _UCC_COMMIT=20eae37090a4ce1b32bcce6144ccad0b49943e0b
+if [[ "$image" == *rocm* ]]; then
+  _UCX_COMMIT=cc312eaa4655c0cc5c2bcd796db938f90563bcf6
+  _UCC_COMMIT=0c0fc21559835044ab107199e334f7157d6a0d3d
+fi
```
Do we really care about about UCC?

jeffdaily replied:
This was added blindly for parity and the assumption that some benchmark was needing it.

huydhn reviewed (approved):
LGTM! Please fix lint before landing

BLOrange-AMD committed:
Fixed lint

jeffdaily committed:
update mi300 runner labels

jeffdaily committed:
swap fbgemm torchrec build order for rocm to avoid fbgemm-gpu-nightly

jeffdaily committed:
doctr models failing to run

jeffdaily committed:
more flaky benchmark models

BLOrange-AMD committed:
Updated flaky benchmark models

jeffdaily committed:
models missing from dynamo_eager_torchbench_training.csv

jeffdaily committed:
more expected accuracy updates

jeffdaily committed:
even more expected accuracy updates

jeffdaily committed:
even more expected accuracy updates, again

BLOrange-AMD commented:
@pytorchbot rebase

jeffdaily commented:
@pytorchbot merge -f "might still have some flaky mi300 benchmark behavior, but no point in delaying merge further because even cuda and cpu accuracy checks are failing"
